docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,348 @@
+```
+
+### oras attach
+
+Attach two files to an image and show the descriptor metadata of the attached files in JSON format.
+
+```bash
+oras attach $REGISTRY/$REPO:$TAG --artifact-type example/vul-scan vul-report.json --artifact-type example/sbom sbom.spdx --format json
+```
+
+```json
+{
+    "Files" : [
+        {
+  "Ref": "localhost:5000/pipe/demo@sha256:5a23319624a3cea05aea5f9dfaf716fba1e7edf8c60d8389af35cebd6f605d30",
+  "MediaType": "application/vnd.oci.image.manifest.v1+json",
+  "Digest": "sha256:5a23319624a3cea05aea5f9dfaf716fba1e7edf8c60d8389af35cebd6f605d30",
+  "Size": 939,
+  "Annotations": {
+        "org.opencontainers.image.title": "vul-report.json"
+      }
+        },
+  {
+  "Ref": "localhost:5000/pipe/demo@sha256:5a23319624a3cea05aea5f9dfaf716fba1e7edf8c60d8389af35cebd6f605333",
+  "MediaType": "application/vnd.oci.image.manifest.v1+json",
+  "Digest": "sha256:5a23319624a3cea05aea5f9dfaf716fba1e7edf8c60d8389af35cebd6f605333",
+  "Size": 929,
+  "Annotations": {
+        "org.opencontainers.image.title": "sbom.spdx"
+      }
+  }
+    ]
+}
+```
+
+### oras push
+
+Push two files to a repository and show the descriptor of the pushed files in pretty JSON format.
+
+```bash
+oras push $REGISTRY/$REPO:$TAG  --format json
+```
+
+```json
+"Ref": "localhost:5000/pipe/demo@sha256:80da7a36f42ab62eeac5382e99fd203149749cf2a861167ada620075e4e6edd4",
+"Files": [
+    {
+      "Ref": "localhost:5000/oras@sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd186222",
+      "MediaType": "application/vnd.oci.image.layer.v1.tar",
+      "Digest": "sha256:6cb759c4296e67e35b0367f3c0f51dfdb776a0c99a45f39d0476e43d82696d65",
+      "Size": 14477,
+      "Annotations": {
+        "org.opencontainers.image.title": "sbom.spdx"
+      }
+    },
+    {
+      "Ref": "localhost:5000/oras@sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd186222",
+      "MediaType": "application/vnd.oci.image.layer.v1.tar",
+      "Digest": "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+      "Size": 0,
+      "Annotations": {
+        "org.opencontainers.image.title": "hello.txt"
+      }
+    }
+  ]
+```
+
+### oras manifest fetch
+
+Fetch a manifest and filter out its reference and media type in standard output:
+
+```bash
+oras manifest fetch $REGISTRY/$REPO:$TAG --format '{{.Ref}}', '{{.Config.mediaType}}'
+```
+
+```json
+localhost:5000/oras@sha256:7414904f07f515f48fe4afeaf876e3151039a81e7177b9c66e9e7ed6dd18669b, application/vnd.oci.empty.v1+json
+```
+
+### oras discover
+
+Discover an artifact's referrers. The default output should be listed in a tree view.
```

Comment:
Let's call out that the default is `--format tree`

Reply:
Updated